msgs/include/simple_msgs/pose_stamped.h

```diff
@@ -70,7 +71,10 @@ class PoseStamped : public GenericMessage {
   /**
    * @brief Returns true if lhs is equal to rhs, false otherwise.
    */
-  inline bool operator==(const PoseStamped& rhs) const { return (pose_ == rhs.pose_ && header_ == rhs.header_); }
+  inline bool operator==(const PoseStamped& rhs) const {
+    std::lock_guard<std::mutex> lock{mutex_};
```

Comment:
should also lock the rhs here, also using `std::lock` as above

Reply:
👍